src/metabase/lib/metric.cljc

```diff
@@ -28,6 +32,18 @@
         lib.convert/->pMBQL
         (lib.util/query-stage -1)))))
 
+(defmethod lib.ref/ref-method :metadata/metric
+  [{:keys [id], :as metric-metadata}]
+  (let [effective-type (or (:effective-type metric-metadata)
+                           (:base-type metric-metadata)
+                           (when-let [[aggregation] (not-empty (:aggregation (metric-definition metric-metadata)))]
+                             (let [ag-effective-type (lib.schema.expression/type-of aggregation)]
+                               (when (isa? ag-effective-type :type/*)
+                                 ag-effective-type))))
+        options (cond-> {:lib/uuid (str (random-uuid))}
+                  effective-type (assoc :effective-type effective-type))]
```

Comment:
It actually works ok, for type calculation if a Metric doesn't have `:effective-type` it's now just returning `::expression/type.unknown` which means it's allowed anywhere as far as the schema is concerned. I don't think we're likely to actually have effective-type or base-type for a Metric since it's not something we record in the application database